### PR TITLE
Added Swift wrappers for push notification methods

### PR DESCRIPTION
**Pull request by jguz-pubnub**

### CODE REVIEW

File: pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/Endpoint.ios.kt [hunk 1]
```diff
@@ -19,3 +23,7 @@ fun <T> Consumer<Result<T>>.onFailureHandler(mapper: (NSError?) -> Throwable = {
         accept(Result.failure(mapper(error)))
     }
 }
+
+fun String.toNSData(): NSData? {
```
Comment (jguz-pubnub) [inline pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/Endpoint.ios.kt:27]: What's the best place for helper methods like this?

File: pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/push/AddChannelsToPush.ios.kt [hunk 2]
```diff
@@ -25,17 +23,15 @@ open class AddChannelsToPushImpl(
     private val deviceId: String
 ): AddChannelsToPush {
     override fun async(callback: Consumer<Result<PNPushAddChannelResult>>) {
-        val deviceIdData = (deviceId as NSString).dataUsingEncoding(NSUTF8StringEncoding)
-
-        if (deviceIdData != null) {
+        deviceId.toNSData()?.let { data: NSData ->
```
Comment (jguz-pubnub) [inline pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/push/AddChannelsToPush.ios.kt:26]: Is something like this ok?

File: pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/Endpoint.ios.kt (hunk 1 above)
Comment (wkal-pubnub) [reply pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/Endpoint.ios.kt:27]: you could create a file in `com.pubnub.kmp` package, for example `converters.kt`, and put some top-level functions there

**Review by wkal-pubnub — APPROVED** [review]

File: pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/push/RemoveAllPushChannelsForDevice.ios.kt [hunk 3]
```diff
@@ -1,9 +1,36 @@
...
 
 /**
  * @see [PubNub.removeAllPushNotificationsFromDeviceWithPushToken]
  */
-actual interface RemoveAllPushChannelsForDevice : Endpoint<PNPushRemoveAllChannelsResult>
\ No newline at end of file
+actual interface RemoveAllPushChannelsForDevice : Endpoint<PNPushRemoveAllChannelsResult>
+
+@OptIn(ExperimentalForeignApi::class)
+open class RemoveAllPushChannelsForDeviceImpl(
```
Comment (marcin-cebo) [inline pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/push/RemoveAllPushChannelsForDevice.ios.kt:21]: I am wondering if "open" is required here?

**Review by marcin-cebo — APPROVED** [review]

File: pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/push/RemoveAllPushChannelsForDevice.ios.kt (hunk 3 above)
Comment (wkal-pubnub) [reply pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/push/RemoveAllPushChannelsForDevice.ios.kt:21]: good point, it's probably not required?

Comment (jguz-pubnub) [reply pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/push/RemoveAllPushChannelsForDevice.ios.kt:21]: I'll check it and remove all occurrences I've written so far in the next PR if it turns out that `open` is not required

**Pull request merged**